test/JDBC/expected/routines_definition.out

```diff
@@ -0,0 +1,669 @@
+~~END~~
+
+
+select ROUTINE_TYPE, DATA_TYPE, CHARACTER_MAXIMUM_LENGTH  from information_schema_tsql.routines where SPECIFIC_NAME NOT LIKE 'xp%';
+go
+~~START~~
+nvarchar#!#nvarchar#!#int
+FUNCTION#!#USER-DEFINED#!#<NULL>
+~~END~~
+
+
+#char
+create procedure test_char(@ch char)
+AS
+BEGIN
+	set @ch ='c';
+	SELECT @ch as 's';
+END;
+go
+
+select tsql_get_functiondef(oid) from pg_proc where proname='test_char';
+go
+~~START~~
+text
+CREATE PROCEDURE dbo.test_char(@ch char) AS BEGIN<newline>	set @ch ='c';<newline>	SELECT @ch as 's';<newline>END;
+~~END~~
+
+
+select CHARACTER_OCTET_LENGTH, COLLATION_NAME  from information_schema_tsql.routines where SPECIFIC_NAME NOT LIKE 'xp%';
+go
+~~START~~
+int#!#nvarchar
+<NULL>#!#<NULL>
+<NULL>#!#<NULL>
+~~END~~
+
+
+drop procedure test_char;
+go
+
+#tinyint, float and bigint
+create procedure test_ti(@a tinyint OUTPUT, @b BIGINT, @c float )
+AS
+BEGIN
+	set @a=79;
+	select @b=19;
+	SELECT @c * 20 +1000;
+END;
+go
+
+select tsql_get_functiondef(oid) from pg_proc where proname='test_ti';
+go
+~~START~~
+text
+CREATE PROCEDURE dbo.test_ti(@a tinyint OUTPUT , @b bigint, @c float) AS BEGIN<newline>	set @a=79;<newline>	select @b=19;<newline>	SELECT @c * 20 +1000;<newline>END;
+~~END~~
+
+
+select ROUTINE_BODY, ROUTINE_DEFINITION  from information_schema_tsql.routines where SPECIFIC_NAME NOT LIKE 'xp%' ORDER BY ROUTINE_DEFINITION;
+go
+~~START~~
+nvarchar#!#nvarchar
+EXTERNAL#!#CREATE FUNCTION dbo.test_dec(@quantity int, @list_price decimal(10,2), @discount decimal(4,2)) RETURNS decimal(10,2) AS BEGIN<newline>    RETURN @quantity * @list_price * (1 - @discount);<newline>END;
+EXTERNAL#!#CREATE PROCEDURE dbo.test_ti(@a tinyint OUTPUT , @b bigint, @c float) AS BEGIN<newline>	set @a=79;<newline>	select @b=19;<newline>	SELECT @c * 20 +1000;<newline>END;
+~~END~~
+
+
+drop procedure test_ti;
+go
+
+#numeric
+create procedure test_num(@a numeric(20,6) OUTPUT)
+AS
+BEGIN
+	set @a = 65;
+	SELECT test_dec(23,60.76,43.88);
+	
+END;
+go
+
+select tsql_get_functiondef(oid) from pg_proc where proname='test_num';
+go
+~~START~~
+text
+CREATE PROCEDURE dbo.test_num(@a numeric(20,6) OUTPUT ) AS BEGIN<newline>	set @a = 65;<newline>	SELECT test_dec(23,60.76,43.88);<newline>	<newline>END;
+~~END~~
+
+
+select NUMERIC_PRECISION, NUMERIC_PRECISION_RADIX, NUMERIC_SCALE, ROUTINE_BODY, ROUTINE_DEFINITION  from information_schema_tsql.routines where SPECIFIC_NAME NOT LIKE 'xp%' ORDER BY ROUTINE_DEFINITION;
+go
+~~START~~
+smallint#!#smallint#!#smallint#!#nvarchar#!#nvarchar
+<NULL>#!#10#!#<NULL>#!#EXTERNAL#!#CREATE FUNCTION dbo.test_dec(@quantity int, @list_price decimal(10,2), @discount decimal(4,2)) RETURNS decimal(10,2) AS BEGIN<newline>    RETURN @quantity * @list_price * (1 - @discount);<newline>END;
+<NULL>#!#<NULL>#!#<NULL>#!#EXTERNAL#!#CREATE PROCEDURE dbo.test_num(@a numeric(20,6) OUTPUT ) AS BEGIN<newline>	set @a = 65;<newline>	SELECT test_dec(23,60.76,43.88);<newline>	<newline>END;
```

Comment:
Ideally we should display the accurate values for precision and scales incase of numeric and decimal datatype.

Reply:
correct! working on it.